subsys/bluetooth/audio/ascs.c

```diff
@@ -873,35 +844,37 @@ static void ase_disable(struct bt_ascs_ase *ase)
 
 static void disconnected(struct bt_conn *conn, uint8_t reason)
 {
-	int i;
-
-	BT_DBG("");
+	struct bt_ascs *session = &sessions[bt_conn_index(conn)];
 
-	for (i = 0; i < CONFIG_BT_MAX_CONN; i++) {
-		struct bt_ascs *ascs = &sessions[i];
+	if (session->conn == NULL) {
+		return;
+	}
 
-		if (ascs->conn != conn) {
-			continue;
-		}
+	bt_conn_unref(session->conn);
+	session->conn = NULL;
 
-		/* Release existing ASEs if not bonded */
-		ascs_clear(ascs);
+	for (size_t i = 0U; i < ARRAY_SIZE(session->ases); i++) {
+		struct bt_audio_stream *stream = session->ases[i].ep.stream;
 
-		bt_conn_unref(ascs->conn);
-		ascs->conn = NULL;
+		if (stream != NULL && stream->conn != NULL) {
+			bt_conn_unref(stream->conn);
+			stream->conn = NULL;
+		}
+	}
 
-		for (size_t i = 0U; i < ARRAY_SIZE(ascs->ases); i++) {
-			struct bt_audio_stream *stream = ascs->ases[i].ep.stream;
+	for (size_t i = 0; i < ASE_COUNT; i++) {
```

Comment:
both loops are iterating over same things,  is this on purpose?    (btw I like use of array_size better for this purpose)